feature/experimental/system/gnmi/benchmarking/ate_tests/drained_configuration_convergence_time/drained_configuration_convergence_time_bgp_test.go

```diff
@@ -0,0 +1,267 @@
+					AfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Prefixes()
+				gotSent := gnmi.Get(t, dut, prefixesv4.Sent().State())
+				switch {
+				case gotSent == setup.RouteCount:
+					t.Logf("Prefixes sent from ingress port are learnt at ATE dst port : %v are %v", setup.ATEIPList[ap.ID()].String(), setup.RouteCount)
+					break prefixLoop
+				case repeat > 0 && gotSent < setup.RouteCount:
+					t.Logf("All the prefixes are not learnt , wait for 5 secs before retry.. got %v, want %v", gotSent, setup.RouteCount)
+					time.Sleep(time.Second * 5)
+				case repeat == 0 && gotSent < setup.RouteCount:
+					t.Errorf("sent prefixes from DUT to neighbor %v is mismatch: got %v, want %v", setup.ATEIPList[ap.ID()].String(), gotSent, setup.RouteCount)
+				}
+			}
+
+			rib := at.NetworkInstance(ap.Name()).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "0").Bgp().Rib()
+			prefixPath := rib.AfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Ipv4Unicast().
+				NeighborAny().AdjRibInPre().RouteAny().WithPathId(0).Prefix()
+			pref := gnmi.GetAll(t, ate, prefixPath.State())
+			asPath := gnmi.GetAll(t, ate, rib.AttrSetAny().AsSegmentAny().State())
+
+			var gotASPath []uint32
+			for _, v := range asPath {
+				gotASPath = append(gotASPath, v.GetMember()...)
+			}
+			if diff := cmp.Diff(wantASPath, gotASPath); diff != "" {
+				t.Errorf("obtained AS path on ATE is not as expected, got %v, want %v, prefixes %v", gotASPath, wantASPath, pref)
+			}
+		}
+	})
+
+	// End the timer and calculate time.
+	elapsed := time.Since(start)
+	t.Logf("Duration taken to apply as path prepend policy is  %v", elapsed)
+}
+
+// verifyBGPSetMED is to Validate MED attribute using bgp rib telemetry on ATE.
+func verifyBGPSetMED(t *testing.T, dut *ondatra.DUTDevice, ate *ondatra.ATEDevice) {
+
+	dutPolicyConfPath := gnmi.OC().NetworkInstance(*deviations.DefaultNetworkInstance).
+		Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp().
+		PeerGroup(setup.PeerGrpName).ApplyPolicy().ExportPolicy()
+
+	// TODO: Below code will be uncommented once configuring MED in DUT as referred in below issue is supported.
+	// Ref: https://github.com/openconfig/featureprofiles/issues/759
+	// Build wantSetMED to compare the diff.
+	// var wantSetMED []uint32
+	// for i := 0; i < setup.RouteCount; i++ {
+	// wantSetMED = append(wantSetMED, bgpMED)
+	// }
+
+	// Start the timer.
+	start := time.Now()
+	gnmi.Replace(t, dut, dutPolicyConfPath.Config(), []string{setMEDPolicy})
+
+	t.Run("BGP-MED-Verification", func(t *testing.T) {
+		// TODO: Below code will be uncommented once SetMED is supported.
+		// Ref: https://github.com/openconfig/featureprofiles/issues/759
+		// at := gnmi.OC()
+		for _, ap := range ate.Ports() {
+			if ap.ID() == "port1" {
+				continue
+			}
+
+			statePath := gnmi.OC().NetworkInstance(*deviations.DefaultNetworkInstance).
+				Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+		prefixLoop:
```

Comment:
Since this code is repeated, consider factoring this into something common whereby the check here is something like `isConverged`?

Reply:
updated, please check. Thank you